sound/soc/sof/topology.c

```diff
@@ -1720,7 +1722,6 @@ static int sof_link_load(struct snd_soc_component *scomp, int index,
 	hw_config = &cfg->hw_config[0];
 
 	config.hdr.cmd = SOF_IPC_GLB_DAI_MSG | SOF_IPC_DAI_CONFIG;
-	config.id = le32_to_cpu(hw_config->id);
```

Comment:
can you clarify if config.id is no longer used at all?
I would like to understand how the matching with the machine driver is done, we are using a .index field in the BE definitions and I can't connect the dots with this patch. Are we saying there is an absolute index for all BEs/DAI, and then a relative one for each DAI type?

Reply:
@plbossart 
This filed should be naming as **index**.
The backend dai here is for firmware only, why should be handle with machine driver?
All these filed I change is only topology vendor data used for IPC binding and Firmware DAI finding.

**These Index is none related with BEs/DAI.**

My first patch is to change the name for all structure.
Could you have a look at https://github.com/thesofproject/sof/pull/140
to see what this name changing will influence the firmware side

Actually I was confused by all these id and index in topology, I can see we have some `id = index` or `index = id` code in kernel. So I take @ranj063 suggestion to rename the filed to be **dai_index** to make it to be the SOF_TKN_DAI_INDEX value

Reply:
@plbossart a minor correction in your statement:

"The firmware only needs to know about the dai_index (relative to the daI_type). The machine driver and BE matching only care about the dai_link ID (which is absolute/linear across all DAI types ). The topology provides both dai_index and link_ID."

Reply:
I call it ID because it doesnt necessarily have to be linear although a linear progression in the machine driver would be nice to have.

Reply:
@plbossart 
actually I still did not get what id and index in what Section is useful in topology.
We will find more cases to make these filed useful.

The id in SectionBE used to be set to be 0 for all topology, but later we find it will be useful if there are too many BE DAIs.

So in this PR, we find more id and index naming issue. We need to have a clean up about naming first. Misuse name will make people confusing even when try to explain things.